### PR TITLE
Impersonate

### PR DESCRIPTION
**Pull request by BohdanBusko**
**STATUS**
READY

**DESCRIPTION**

**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs

### CODE REVIEW

File: app/controllers/admin/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Admin::UsersController < Admin::AdminController
+  before_action :require_admin!
```
Comment (romaslmd) [inline app/controllers/admin/users_controller.rb]: no need for double checks

File: spec/controllers/admin/users_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require 'rails_helper'
+
+RSpec.describe Admin::UsersController, type: :controller do
+  render_views
+
+  let!(:user_admin) { create(:user, :admin) }
+  let!(:user) { create(:user) }
+
+  describe 'admin impersonate'  do
```
Comment (romaslmd) [inline spec/controllers/admin/users_controller_spec.rb]: correct describss and contexts!

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -63,4 +63,9 @@
     it { should validate_length_of(:first_name).is_at_most(250) }
     it { should validate_length_of(:last_name).is_at_most(250) }
   end
+
+  context 'user admin' do
```
Comment (romaslmd) [inline spec/models/user_spec.rb]: why context?
this should be a desribe "factory", context :admin trait or something

File: app/controllers/admin/admin_controller.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class Admin::AdminController < ApplicationController
+  impersonates :user
+
+  before_action :authenticate_user!
+
+  private
+
+  def require_admin!
```
Comment (igorkasyanchuk) [inline app/controllers/admin/admin_controller.rb:8]: no no no!!!

DON"T DO THIS
DON"T OVERRIDE DEVISE METHODS

think how to fix that issue

File: app/models/user.rb [hunk 5]
```diff
@@ -91,4 +91,8 @@ def watching?(task)
   def with_avatar?
     avatar&.attachment&.blob&.persisted?
   end
+
+  def admin?
+    self.role == 'admin'
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: move to contstant

**Review by igorkasyanchuk — APPROVED** [review]

**Pull request merged**